### DISCUSSION TITLE
ObjectFactory Create method in generic abstract base class works for BusinessBase but not CommandBase

### DISCUSSION THREAD

**Discussion by kellyethridge** [MarimerLLC/csla#1704] [Questions] [answered]
Hello,

I have an `ObjectFactory` that is an abstract generic repository base class for `BusinessBase` derived classes. In this base class I have a default virtual create method. This has been working as expected for years. Now I'm trying to do the same thing for a generic abstract `ObjectFactory` for `CommandBase` derived classes. Unfortunately, the default create method fails with a `TargetParameterCountException` being thrown. There are no parameters being passed for the default implementation. I would have expected this to work just like the `BusinessBase` version of the factory class.

My simple default `Create` method in the factory looks like this:
```
        [RunLocal]
        public virtual Task<object> Create()
        {
            object cmd = new T();
            return Task.FromResult(cmd);
        }
```
This is basically the same as the `BusinessBase` version.

I call the method using:
```
var cmd = await DataPortal.CreateAsync<MyCommandClass>();
```
So I'm not sure why that particular exception is being thrown. 

I temporarily converted the class from a `CommandBase` to a `BusinessBase` and the create method started working as expected. I've tried the sync implementation as well, still with no luck.

Thank you.

**Comment by rockfordlhotka** [comment, marked as answer]
The object factory model is unchanged from earlier versions of CSLA - I've implemented the new capabilities only for the encapsulated models so far.
